tests/QueryCompatibleInputMiddlewareTest.php

```diff
@@ -0,0 +1,256 @@
+<?php
+namespace Aws\Test;
+
+use Aws\AwsClient;
+use Aws\Command;
+use Aws\HandlerList;
+use Aws\Api\Service;
+use Aws\Middleware;
+use Aws\QueryCompatibleInputMiddleware;
+use GuzzleHttp\Psr7\Request;
+use Yoast\PHPUnitPolyfills\TestCases\TestCase;
+
+/**
+ * @covers \Aws\QueryCompatibleInputMiddleware
+ */
+class QueryCompatibleInputMiddlewareTest extends TestCase
+{
+    /**
+     * @dataProvider getInputs()
+     *
+     * @param $inputParam
+     * @param $inputValue
+     * @param $expected
+     * @param $type
+     */
+    public function testEmitsWarning($inputParam, $inputValue, $expected, $type)
+    {
+        $this->expectWarning();
+        $this->expectWarningMessage(
+            "The provided type for `${inputParam}` value was `"
+            . (gettype($inputValue) === 'double' ? 'float' : gettype($inputValue))
+            . "`. The modeled type is `{$type}`."
+        );
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service);
+        $command = $client->getCommand(
+            'FooOperation',
+            [$inputParam => $inputValue]
+        );
+
+        $list = new HandlerList();
```

Comment:
Why manually construct a `HandlerList`? Can't the handler be configured on the test client, then these tests use the mocked client?

Reply:
Yeah, this could definitely be done in a more straightforward way.  I borrowed some of this from another middleware test class and didn't give it a close enough look.